--- a/client/pool-buffer.c
+++ b/client/pool-buffer.c
@@ -1,4 +1,4 @@
-#define _POSIX_C_SOURCE 200809
+#define _GNU_SOURCE
 #include <assert.h>
 #include <cairo.h>
 #include <errno.h>
@@ -15,7 +15,7 @@
 #include "pool-buffer.h"
 #include "util.h"
 
-static int anonymous_shm_open(void) {
+inline static int anonymous_shm_open(void) {
 	int retries = 100;
 
 	do {
@@ -55,7 +55,7 @@
 	uint32_t stride = width * 4;
 	size_t size = stride * height;
 
-	int fd = anonymous_shm_open();
+	int fd = memfd_create("sway", MFD_CLOEXEC);
 	if (fd == -1) {
 		return NULL;
 	}
