--- a/main.c
+++ b/main.c
@@ -35,24 +35,6 @@
 }
 
 static char *
-get_config_path_user_config(void)
-{
-    struct passwd *passwd = getpwuid(getuid());
-    if (passwd == NULL) {
-        LOG_ERRNO("failed to lookup user");
-        return NULL;
-    }
-
-    const char *home_dir = passwd->pw_dir;
-    LOG_DBG("user's home directory: %s", home_dir);
-
-    int len = snprintf(NULL, 0, "%s/.config/yambar/config.yml", home_dir);
-    char *path = malloc(len + 1);
-    snprintf(path, len + 1, "%s/.config/yambar/config.yml", home_dir);
-    return path;
-}
-
-static char *
 get_config_path_xdg(void)
 {
     const char *xdg_config_home = getenv("XDG_CONFIG_HOME");
@@ -76,8 +58,8 @@
     free(config);
 
     /* 'Default' XDG_CONFIG_HOME */
-    config = get_config_path_user_config();
-    if (config != NULL && stat(config, &st) == 0 && S_ISREG(st.st_mode))
+    config = strdup("/etc/yambar/config.yml");
+    if (access(config, F_OK) == 0)
         return config;
     free(config);
 
